src/Microsoft.Azure.WebJobs.Host/Bindings/ConverterManager.cs

```diff
@@ -4,23 +4,85 @@
 using System;
 using System.Collections;
 using System.Collections.Generic;
+using System.Reflection;
 using System.Text;
 using Microsoft.Azure.WebJobs.Host.Bindings;
 using Newtonsoft.Json.Linq;
 
 namespace Microsoft.Azure.WebJobs
-{
+{   
     // Concrete implementation of IConverterManager
     internal class ConverterManager : IConverterManager
     {
         // Map from <TSrc,TDest> to a converter function. 
-        private Dictionary<string, object> _funcsWithAttr = new Dictionary<string, object>();
-        
+        // (Type) --> FuncConverter<object, TAttribute, object>
+        private readonly Dictionary<string, object> _funcsWithAttr = new Dictionary<string, object>();
+
+        private readonly List<Entry> _openConverters = new List<Entry>();
+
+        public static readonly IConverterManager Identity = new IdentityConverterManager();
+
         public ConverterManager()
         {
             this.AddConverter<byte[], string>(DefaultByteArray2String);
         }
 
+        private void AddOpenConverter<TSrc, TDest, TAttribute>(
+          Func<Type, Type, Func<object, object>> converterBuilder)
+          where TAttribute : Attribute
+        {
+            var entry = new Entry
+            {
+                SourceCheck = GetChecker<TSrc>(),
+                DestCheck = GetChecker<TDest>(),
+                Attribute = typeof(TAttribute),
+                Builder = converterBuilder
+            };
+            this._openConverters.Add(entry);
+        }
+         
+        // Given a Type, get a predicate for determining if a given type matches it. 
+        // This handles OpenTypes.        
+        private static Func<Type, bool> GetChecker<TOpenType>()
+        {
+            var t = typeof(TOpenType);
+
+            if (!typeof(OpenType).IsAssignableFrom(t))
+            {
+                // If not an OpenType, then must be an exact match. 
+                return (type) => (type == t);
+            }
+
+            if (t == typeof(OpenType))
+            {
+                return (type) => true;
+            }
+            var method = t.GetMethod("IsValid", BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic);
+            if (method == null)
+            {
+                throw new InvalidOperationException("Type " + t.FullName + "' is derived from " + typeof(OpenType).Name + " but is missing an " 
```

Comment:
what's that? 

---
In reply to: [96106526](https://github.com/Azure/azure-webjobs-sdk/pull/967#discussion_r96106526) [](ancestors = 96106526)